cirq-core/cirq/experiments/xeb_sampling.py

```diff
@@ -276,18 +274,16 @@ def _execute_sample_2q_xeb_tasks_in_batches(
     run_batch = _SampleInBatches(
         sampler=sampler, repetitions=repetitions, combinations_by_layer=combinations_by_layer
     )
-    with ThreadPoolExecutor(max_workers=2) as pool:
-        futures = [pool.submit(run_batch, task_batch) for task_batch in batched_tasks]
 
-        records = []
-        with progress_bar(total=len(batched_tasks) * batch_size) as progress:
-            for future in concurrent.futures.as_completed(futures):
-                new_records = future.result()
-                if dataset_directory is not None:
-                    os.makedirs(f'{dataset_directory}', exist_ok=True)
-                    protocols.to_json(new_records, f'{dataset_directory}/xeb.{uuid.uuid4()}.json')
-                records.extend(new_records)
-                progress.update(batch_size)
+    records = []
+    with progress_bar(total=len(batched_tasks) * batch_size) as progress:
+        for task in batched_tasks:
+            new_records = run_batch(task)
+            if dataset_directory is not None:
+                os.makedirs(f'{dataset_directory}', exist_ok=True)
+                protocols.to_json(new_records, f'{dataset_directory}/xeb.{uuid.uuid4()}.json')
```

Comment:
Would the order of records be useful here?
If so, it would be better to write json at the end when they are all collected in the `records` list or at least add an enumeration suffix to the filename.

Reply:
without threading the records are written in the same order as their inputs ... using threads with as_completed (the currently) behaviour wrote them in the order they were completed (so randomly) .... adding the suffix can work now but it will break when I reintroduce the optional suffixes ... eitherway this seems to be more of debug feature than anything else

Reply:
Sounds good, thanks.

@eliottrosenberg - can you chime in if this debugging feature is still needed?

If not, we should consider removing it in a separate PR.

Reply:
I'm not following at this level of detail; I am just a user here :)

Reply:
fyi, this part of the library was written 3+ years ago

Reply:
I see, let's then leave it as is.